<!-- Starts up everything needed to run the example listener/talker nodes. -->
<launch>
  <!-- Arguments to allow changing common parameters from the command line or when including this file from other launch files. -->
  <arg name="a" default="1"/>
  <arg name="b" default="2"/>
  <arg name="message" default="hello"/>
  <arg name="rate" default="1.0"/>

  <!-- The talker node will publish a custom message. -->
  <node pkg="debug_motor" type="d_motor" name="d_motor" output= "screen">
    <param name="a" value="$(arg a)"/>
    <param name="b" value="$(arg b)"/>
    <param name="message" value="$(arg message)"/>
    <param name="rate" value="$(arg rate)"/>
    
    <param name="dev" value="/dev/ttyUSB0" type="str" />
	<param name="time_out" value="1000" type="int" />
	<param name="baud_rate" value="115200" type="int" />
	
    <remap from="example" to="chatter"/>
  </node>

  <!-- The listener node subscribes to the custom message that talker is publishing. -->
<!--  <node pkg="node_example" type="listener" name="listener" output="screen" >-->
<!--    <remap from="example" to="chatter"/>-->
<!--  </node>-->

  <include file="$(find debug_motor)/launch/tools.launch"/>
</launch>
